File: src/models/Bracket.js
const mongoose = require('mongoose');

const bracketSchema = new mongoose.Schema({
  tournament: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Tournament',
    required: true,
    unique: true
  },
  matches: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Match'
  }],
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Bracket', bracketSchema);

File: Models/Bracket.cs
namespace TournamentAPI.Models;

public class Bracket
{
    public int Id { get; set; }
    public int TournamentId { get; set; }
    
    // Navigation properties
    public Tournament Tournament { get; set; } = null!;
    public ICollection<Match> Matches { get; set; } = new List<Match>();
}
